tests/unit/amo/components/TestSearchFilters.js

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+
+import SearchFilters, {
+  SearchFiltersBase,
+} from 'amo/components/SearchFilters';
+import { searchStart } from 'core/actions/search';
+import { ADDON_TYPE_EXTENSION, OS_LINUX } from 'core/constants';
+import { convertFiltersToQueryParams } from 'core/searchUtils';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import {
+  createFakeEvent,
+  createStubErrorHandler,
+  getFakeI18nInst,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  const fakeEvent = createFakeEvent();
+  let context;
+
+  function render({
+    filters = {},
+    pathname = '/search/',
+    ...props
+  } = {}) {
+    const errorHandler = createStubErrorHandler();
+    const { store } = dispatchClientMetadata();
+
+    store.dispatch(searchStart({ errorHandlerId: errorHandler.id, filters }));
+
+    return shallowUntilTarget(
+      <SearchFilters
+        i18n={getFakeI18nInst()}
+        store={store}
+        pathname={pathname}
+        {...props}
+      />, SearchFiltersBase,
+      { shallowOptions: { context } }
+    );
+  }
+
+  beforeEach(() => {
+    context = { router: { push: sinon.stub() } };
```

Comment:
It looks like the only variable you need to declare in the upper scope is `router`.